lib/dialects/abstract/query-generator.js

```diff
@@ -1347,7 +1365,11 @@ var QueryGenerator = {
     if (subQuery) {
       subQueryItems.push('SELECT ' + subQueryAttributes.join(', ') + ' FROM ' + table);
       if (mainTableAs) {
-        subQueryItems.push(' AS ' + mainTableAs);
+        if (this._dialect.supports.as) {
```

Comment:
not very DRY :) 

Perhaps we could abstract it into a function , something like

```js
as: function (str) {
 return (this._dialect.supports.as ? ' AS ' : ' ') + str;
}

subQueryItems.push(this.as(mainTableAs));
```